op-e2e/config/init.go

```diff
@@ -144,69 +171,370 @@ func init() {
+	for _, fork := range forks {
+		l2OOAllocsL2B := decompressGzipFile(path.Join(configPath, fmt.Sprintf("allocs-l2-%s.json.gz", fork)))
+		var l2OOAllocsL2 foundry.ForgeAllocs
+		if err := json.Unmarshal(l2OOAllocsL2B, &l2OOAllocsL2); err != nil {
+			panic(fmt.Errorf("failed to unmarshal L2OO L2 allocs: %w", err))
 		}
+		l2AllocsByType[AllocTypeL2OO][fork] = &l2OOAllocsL2
 	}
-	if missing {
-		log.Warn("allocs file not found, skipping", "allocType", allocType)
-		return
+
+	// Use regular level going forward.
+	oplog.SetGlobalLogHandler(handler)
+}
+
+func initAllocType(root string, allocType AllocType) {
+	artifactsPath := path.Join(root, "packages", "contracts-bedrock", "forge-artifacts")
+	if err := ensureDir(artifactsPath); err != nil {
+		panic(fmt.Errorf("invalid artifacts path: %w", err))
 	}
 
-	l1Allocs, err := foundry.LoadForgeAllocs(l1AllocsPath)
+	loc, err := artifacts.NewFileLocator(artifactsPath)
 	if err != nil {
-		panic(err)
+		panic(fmt.Errorf("failed to create artifacts locator: %w", err))
 	}
-	l1AllocsByType[allocType] = l1Allocs
+
+	lgr := log.New()
+
+	allocModes := []genesis.L2AllocsMode{
+		genesis.L2AllocsHolocene,
+		genesis.L2AllocsGranite,
+		genesis.L2AllocsFjord,
+		genesis.L2AllocsEcotone,
+		genesis.L2AllocsDelta,
+	}
+
 	l2Alloc := make(map[genesis.L2AllocsMode]*foundry.ForgeAllocs)
-	mustL2Allocs := func(mode genesis.L2AllocsMode) {
-		name := "allocs-l2-" + string(mode)
-		allocs, err := foundry.LoadForgeAllocs(filepath.Join(l2AllocsDir, name+".json"))
-		if err != nil {
-			panic(err)
-		}
-		l2Alloc[mode] = allocs
+	var wg sync.WaitGroup
+
+	// Corresponds with the Deployer address in cfg.secrets
+	pk, err := crypto.HexToECDSA("7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6")
+	if err != nil {
+		panic(fmt.Errorf("failed to parse private key: %w", err))
 	}
-	mustL2Allocs(genesis.L2AllocsHolocene)
-	mustL2Allocs(genesis.L2AllocsGranite)
-	mustL2Allocs(genesis.L2AllocsFjord)
-	mustL2Allocs(genesis.L2AllocsEcotone)
-	mustL2Allocs(genesis.L2AllocsDelta)
+	deployerAddr := crypto.PubkeyToAddress(pk.PublicKey)
+	lgr.Info("deployer address", "address", deployerAddr.Hex())
+
+	for _, mode := range allocModes {
+		wg.Add(1)
+		go func(mode genesis.L2AllocsMode) {
+			defer wg.Done()
+
+			intent := defaultIntent(root, loc, deployerAddr)
+			if allocType == AllocTypeAltDA {
+				intent.Chains[0].DangerousAltDAConfig = genesis.AltDADeployConfig{
+					UseAltDA:                   true,
+					DACommitmentType:           "KeccakCommitment",
+					DAChallengeWindow:          16,
+					DAResolveWindow:            16,
+					DABondSize:                 1000000,
+					DAResolverRefundPercentage: 0,
+				}
+			}
+
+			baseUpgradeSchedule := map[string]any{
+				"l2GenesisRegolithTimeOffset": nil,
+				"l2GenesisCanyonTimeOffset":   nil,
+				"l2GenesisDeltaTimeOffset":    nil,
+				"l2GenesisEcotoneTimeOffset":  nil,
+				"l2GenesisFjordTimeOffset":    nil,
+				"l2GenesisGraniteTimeOffset":  nil,
```

Comment:
This is going to add another place that we have to remember to update when we are adding a new hardfork. Just calling out that it will take some time for engineers to learn that this is another location that needs to be updated. It could be good to have @sebastianst look at this as he has some ideas on how to reduce the boilerplate required for when adding a hardfork